components/channel_header/channel_header.jsx

```diff
@@ -317,7 +318,7 @@ export default class ChannelHeader extends React.Component {
             >
                 <Markdown
                     message={channel.header}
-                    options={headerMarkdownOptions}
+                    options={{...headerMarkdownOptions, channelNamesMap}}
```

Comment:
Creating a new object for options on each render is going to cause the Markdown component to also re-render. I don't know how often ChannelHeader re-renders, but we might want to memoize this if that happens a lot.